src/Paket.Core/PaketConfigFiles/ProjectFile.fs

```diff
@@ -278,6 +278,11 @@ module ProjectFile =
     let private appendMap first second =
         Map.fold (fun state key value -> Map.add key value state) first second
 
+    let inline getPackageIdAttribute (pf:ProjectFile) (node:XmlNode) =
+        node |> getAttribute "Include"
+        |> Option.orElseWith (fun _ -> node |> getAttribute "Update")
+        |> Option.defaultWith (fun _ -> failwithf "project file '%s' contains a reference without 'Include' or 'Update' attribute" pf.FileName)
```

Comment:
that's a weird technique. throwing in defaultWith!? Seriously?

Reply:
What else do you suggest? The only thing it does: Give a better error message in case all other cases fail and `defaultWith` has the signature I need for this here ;)

Reply:
Honestly, I "default" "with" an error message. It doesn't even sound too bad. Is the code better with a `match`?